server/src/main/java/io/deephaven/server/appmode/ApplicationServiceGrpcImpl.java

```diff
@@ -209,7 +210,8 @@ private synchronized void propagateUpdates() {
         updatedFields.clear();
         final FieldsChangeUpdate update = builder.build();
 
-        subscriptions.forEach(sub -> sub.send(update));
+        List<Subscription> subscriptionsToCancel = subscriptions.stream().filter(sub -> sub.send(update)).collect(Collectors.toList());
```

Comment:
filter predicates *should* be stateless ...

Reply:
Yeah - removeIf is so much clearer, thanks for the kick to just do it right.

I almost went with removeIf right away after seeing your comment, then assume that it would be equally "inappropriate" to do so, even though, like Stream.filter with a collect on the end, I know it will traverse every item. Instead I went with 
```
        for (Iterator<Subscription> iterator = subscriptions.iterator(); iterator.hasNext(); ) {
            Subscription subscription = iterator.next();
            if (!subscription.send(update)) {
                iterator.remove();
            }
        }
```
And IJ gave me a "why don't you just use Collection.removeIf" popup. Fixed.